### ISSUE TITLE
Fix Request for HttpClientBuilderWithoutTimeouts missed check and false positive

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#191]
If HttpAsyncClientBuilder with callbacks is used, a violation is reported without any checking.
Example:
```
class DataStoreConfigGood {
    private RestClientBuilder getRestClientBuilder() {
        return RestClient.builder(hosts)
                .setHttpClientConfigCallback(this::createHttpClientConfigCallback)
                .setRequestConfigCallback(this::createRequestConfigCallbackGood);
    }
    private HttpAsyncClientBuilder createHttpClientConfigCallback(final HttpAsyncClientBuilder clientBuilder) {
        clientBuilder
                .setMaxConnPerRoute(20)
                .setMaxConnTotal(20)
                .disableConnectionState();
        return clientBuilder;
    }
    private RequestConfig.Builder createRequestConfigCallbackGood(RequestConfig.Builder builder) {
        return builder
                .setConnectTimeout(200)
                .setConnectionRequestTimeout(200)
                .setSocketTimeout(4000);
    }
}

class DataStoreConfigBad {
    private RestClientBuilder getRestClientBuilder() {
        return RestClient.builder(hosts)
                .setHttpClientConfigCallback(this::createHttpClientConfigCallback)
                .setRequestConfigCallback(this::createRequestConfigCallbackBad);
    }
    private HttpAsyncClientBuilder createHttpClientConfigCallback(final HttpAsyncClientBuilder clientBuilder) { // bad
        clientBuilder
                .setMaxConnPerRoute(20)
                .setMaxConnTotal(20)
                .disableConnectionState();
        return clientBuilder;
    }
    private RequestConfig.Builder createRequestConfigCallbackBad(RequestConfig.Builder builder) {
        return builder
                .setConnectTimeout(200) 
                .setSocketTimeout(4000); // one timeout missing
    }
}

```
